torch/csrc/autograd/python_variable.cpp

```diff
@@ -694,20 +694,35 @@ static PyObject* THPVariable_make_wrapper_subclass(
     AutoDispatchBelowADInplaceOrView guard{}; // TODO: Remove.
     tracer::impl::NoTracerDispatchMode tracer_guard{};
 
-    // We shouldn't need storage
-    Storage storage{Storage::use_byte_size_t{}, 0, at::DataPtr{}};
+    // We use storages **only** to track aliasing of subclasses during tracing.
+    // The actual data pointers are not valid.
+    Storage storage{
+        Storage::use_byte_size_t{},
+        0,
+        at::DataPtr{},
+        /*allocator=*/c10::GetAllocator(c10::kMeta),
+        /*resizeable=*/true};
 
     tensor = at::detail::make_tensor<TensorImpl>(
         std::move(storage), options.computeDispatchKey(), options.dtype());
 
     auto sym_sizes = r.symintlist(1);
-    auto sym_strides = r.symintlist(2);
+    auto sym_strides = r.symintlistOptional(2);
     auto sym_storage_offset = r.toSymIntOptional(3);
 
     TensorImpl* tensor_impl = tensor.unsafeGetTensorImpl();
 
-    tensor_impl->set_sizes_and_strides(
-        sym_sizes, sym_strides, sym_storage_offset.value_or(0));
+    if (sym_strides.list.has_value()) {
+      tensor_impl->set_sizes_and_strides(
+          sym_sizes, sym_strides.list.value(), sym_storage_offset.value_or(0));
+    } else {
+      tensor_impl->generic_set_sizes_contiguous(sym_sizes);
+      if (sym_storage_offset.has_value()) {
+        // Replicating what TensorImpl does with storage_offset:
+        // https://github.com/pytorch/pytorch/blob/main/c10/core/TensorImpl.cpp#L1144
+        tensor_impl->set_storage_offset(sym_storage_offset->as_int_unchecked());
```

Comment:
This seems bad. You don't actually know that sym storage offset is a plain int

Reply:
Right. I copied this because this is currently how `TensorImpl` sets the storage offset, when you pass in a `SymInt` as your storage offset (originally added a year ago [here](https://github.com/pytorch/pytorch/pull/84806/files#diff-052bd9150ef8e09289ddf644b5a6830ede49207201cd41728f6d7cc6d9cead94R899)). It seems... not quite right, but no worse than how we handle symbolic storage offsets normally today? Maybe I'm missing something

Reply:
In the quoted code, it is OK to use `as_int_unchecked` because it is guarded by a conditional that storage offset must be integer. You don't have that test here!

Reply:
Good catch. In fact, I'm removing these changes - they were left over from when I originally changed this `_make_wrapper_subclass` overload to take an optional `strides` param, but are no longer necessary.